Save Firecracker PID in Jailer's root directory (#4442)
firecracker-microvm/firecracker: docs/jailer.md

### Known limitations

- When passing the --daemonize option to Firecracker without the --new-ns-pid

>   option, the Firecracker process will have a different PID than the Jailer

Reviewer: Isn't this always true when using the daemonize option, regardless of the `--new-pid-ns`?

Author: rephrased the comments to make the issue more clear.

Reviewer 2: +1 to remove "without --new-ns-pid option".

### Known limitations

- When passing the --daemonize option to Firecracker without the --new-ns-pid
  option, the Firecracker process will have a different PID than the Jailer

>   process. As a workaround to get Firecracker PID, the Jailer now stores the PID

Reviewer 2: Using "now" sounds CHANGELOG to me (like "we didn't do A, but now we do"). This is a doc and it should always the current behavior.
```suggestion
  process. As a workaround to get Firecracker PID, the Jailer stores the PID
```

  workaround to get Firecracker PID, the Jailer stores the PID of the child
  process in the jail root directory inside `<exec_file_name>.pid` for all cases

>   regardless of whether the new_pid_ns flag was set. The suggested way to fetch

Reviewer 2: nit: as users shouldn't care the internal flag, it would be better to use the user-facing parameter name here.
```suggestion
  regardless of whether `--new_pid_ns` was provided. The suggested way to fetch
```

>   way to fetch Firecracker process's pid when using the Jailer is to read

Reviewer 2: for consistency
```suggestion
  way to fetch Firecracker PID when using the Jailer is to read
```

>   Firecracker's PID from the `firecracker.pid` file present in the Jailer's root

Reviewer 2: nit: It looks a bit duplicated.
```suggestion
  `firecracker.pid` file present in the Jailer's root
```
